### ISSUE TITLE
Reboot pairwise as PairwiseBorrowing class

### ISSUE THREAD

**Issue by fractaldragonflies** [lexibank/sabor#11]
[John: Copied this plan from @LinguList from our reboottestfiles branch to continue this discussion after completing merge of reboottestfiles.]

Thanks, so we have this first step done now! We can then start to 

1. adjust our tests for the pairwise workflow on this new k-fold data. 
2. discuss how to implement the other three (lingrex, family, SVM) methods

I'd suggest, let us first work on 1, so that we have a workflow where we "train" by picking the best threshold first, and then applying this to get the predictions on the test set. Once this workflow is there, it'll be easier to work on 2. 

Since we want to check automatically for the threshold from the training data, we would best write a Class that extends a LingPy wordlist and has a `train` method that defines which thresholds should be tested. 

```python
class PairwiseBorrowing(LexStat):
    def __init__(self, infile, func, donor, **kw):
        LexStat.__init__(self, infile, **kw)
        self.func = func
        self.donor = donor
    def get_borrowings(self, threshold=0.3, ref="borid"):
        ...
    def train(self, thresholds=[i*0.05 for i in range(20)]):
        for t in thresholds:
            self.get_borrowings(ref="t_{0:.2f}".format(t), threshold=t)
            # then check for accuracy or other metric here and store it
        self.best_t = best_t
    def predict(self, words, languages):
        # function should do the same as inside pairwise, maybe do a specific function inside pairwise to do the prediction here, so we predict for a couple of words and a couple of languages
```

I have not thougth this through 100% but think this should work nicely. I could otherwise also start to write it up.

_Originally posted by @LinguList in https://github.com/lexibank/sabor/pull/10#pullrequestreview-964724757_
